### PR TITLE
Support configurable namespace for docker magic command in notebook

### PR DESCRIPTION
**Pull request by gaoning777**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1473)

### CODE REVIEW

**Comment by gaoning777** [comment]
Changing the default namespace back to "kubeflow" because it will break existing samples. The reason is that the kaniko job depends on user service account that only exists in the kubeflow namespace.

However, I added the support for docker magic to configure the namespace.

File: sdk/python/kfp/notebook/_magic.py [hunk 1]
```diff
@@ -21,18 +21,20 @@
 def docker(line, cell):
   """cell magic for %%docker"""
 
-  if len(line.split()) != 2:
+  if len(line.split()) not in {2,3}:
     raise ValueError("usage: %%docker [gcr.io/project/image:tag] [gs://staging-bucket]")
```
Comment (IronPan) [inline sdk/python/kfp/notebook/_magic.py]: can you update the error message

Comment (gaoning777) [reply sdk/python/kfp/notebook/_magic.py]: good catch. Updated.

**Comment by IronPan** [comment]
Does kaniko have to depend on GCS, or some other gcp services that need service account credential? I think we should avoid that as much as possible. If not, we should just use cloud build.

**Comment by gaoning777** [comment]
Kaniko does not have to depend on GCS. An issue was created to keep track: https://github.com/kubeflow/pipelines/issues/1035. However, using k8s build with configmap is better than cloudbuild IMO such that it works on-prem. 
This PR, though, is about the configurable namespace for the magic docker command in the notebook. We can discuss removing the GCS dependency on that issue. WDYT?

**Comment by IronPan** [comment]
If we can avoid depend on GCS let's do that. It seems the dependency has trickle down to the change for this change. Let's aim to reduce the tech debt and not optimize for specific task. WDYT?

**Comment by gaoning777** [comment]
> If we can avoid depend on GCS let's do that. It seems the dependency has trickle down to the change for this change. 

You have a good point that using a different namespace than 'kubeflow' will not work. At first, the PR was to change the default namespace. Then, the PR was shifted to expose the namespace configuration in docker magic command. 
> Let's aim to reduce the tech debt and not optimize for specific task. WDYT?

Totally agree that we need to prioritize the task a bit higher since we do not want to lock to a certain namespace. 

I'm OK with not merging this PR. BTW, using the configmap takes a bit more time and I will do it when I have the bandwidth. Thanks

**Comment by gaoning777** [comment]
Closed in favor of https://github.com/kubeflow/pipelines/pull/1550

**Pull request closed without merging**